Sources/SwiftDocC/Infrastructure/Symbol Graph/SymbolGraphLoader.swift

```diff
@@ -232,14 +232,24 @@ struct SymbolGraphLoader {
                         newAvailabilityItems.append(macCatalystAvailability)
                     }
 
-                    // If a symbol doesn't have any availability annotation at all
-                    // for a given platform, create a new one just with the
-                    // introduced version so that it shows up in the sidebar.
+                    // If a symbol doesn't have any availability version
+                    // annotation at all for a given platform, create a new
+                    // one just with the introduced version so that it shows
+                    // up in the sidebar.
                     for defaultAvailability in defaultAvailabilities {
+                        // If the symbol does not exists for the given platform
+                        // don't add any default availability.
+                        guard newAvailabilityItems.contains(where: {
+                            $0.domain?.rawValue == defaultAvailability.platformName.rawValue
+                        }) else { continue }
+                        // If the symbol exists for the platform check if it also defines the availability
+                        // version information.
                         let hasAvailabilityForThisPlatform = newAvailabilityItems.contains {
-                            guard let domain = $0.domain else { return false }
-                            return PlatformName(operatingSystemName: domain.rawValue) == defaultAvailability.platformName
+                            // Safe to force unwrap below, we already checked that the domain does exists at this point.
+                            return PlatformName(operatingSystemName: $0.domain!.rawValue) == defaultAvailability.platformName
```

Comment:
what is the difference between platformName and platformName.rawValue? and why we are comparing these two different things in two places.